File: 2507-number-of-common-factors/2507-number-of-common-factors.js
/**
 * @param {number} a
 * @param {number} b
 * @return {number}
 */
var commonFactors = function(a, b) {
    let count = 0;
    let lim = Math.min(a, b); 
    
    for (let i = 1; i <= lim; i++) {
        if (a % i === 0 && b % i === 0) { 
            count++;
        }
    }
    
    return count; 
};

File: 2507-number-of-common-factors/2507-number-of-common-factors.cs
public class Solution {
    public int CommonFactors(int a, int b) {
      int count = 0;
    int lim = Math.Min(a, b); 
    
    for (int i = 1; i <= lim; i++) {
        if (a % i == 0 && b % i == 0) { 
            count++;
        }
    }
    
    return count;
    }
}